simd/src/Kokkos_SIMD_Scalar.hpp

```diff
@@ -532,19 +606,19 @@ template <class T>
 #endif
 
 template <class T>
-[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION T
-reduce(const_where_expression<basic_simd_mask<T, simd_abi::scalar>,
-                              basic_simd<T, simd_abi::scalar>> const& x,
-       T identity_element, std::plus<>) {
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION constexpr T reduce(
+    const_where_expression<basic_simd_mask<T, simd_abi::scalar>,
+                           basic_simd<T, simd_abi::scalar>> const& x,
+    T identity_element, std::plus<>) {
   return static_cast<bool>(x.impl_get_mask())
              ? static_cast<T>(x.impl_get_value())
              : identity_element;
 }
 
 template <class T>
-[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION T
-reduce_max(const_where_expression<basic_simd_mask<T, simd_abi::scalar>,
-                                  basic_simd<T, simd_abi::scalar>> const& x) {
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION constexpr T reduce_max(
+    const_where_expression<basic_simd_mask<T, simd_abi::scalar>,
+                           basic_simd<T, simd_abi::scalar>> const& x) noexcept {
```

Comment:
Another reminder these seem to all be the other way around in the standard (i.e. mask as the second argument)?

Reply:
I'm actually planning on deprecating everything related to `const_where_expression` (copy_[from|to], reduce_[max|min], etc). It's part of `stdx::simd`, but not in c++26 simd.
In P1928, functionalities that previously used `const_where_expression` were replaced.
Let me know what you think.